src/openrct2/ride/Ride.cpp

```diff
@@ -1997,7 +1996,7 @@ int32_t ride_modify(CoordsXYE* input)
 
     if (ride_type_has_flag(ride->type, RIDE_TYPE_FLAG_CANNOT_HAVE_GAPS))
     {
-        if (ride_find_track_gap(&tileElement, &endOfTrackElement))
+        if (ride_find_track_gap(rideIndex, &tileElement, &endOfTrackElement))
```

Comment:
Good point actually

Reply:
Well I still have to pass rideIndex as that is required and it seems Ride does not store its own index, at least it can avoid getting the ride again in the new overload.